deck-core/src/main/kotlin/com/deck/core/entity/TeamChannel.kt

```diff
@@ -1,7 +1,106 @@
 package com.deck.core.entity
 
-import com.deck.common.util.GenericId
+import com.deck.common.util.*
+import com.deck.common.util.Dictionary
+import com.deck.core.entity.impl.DeckRole
+import com.deck.core.entity.misc.ChannelContentType
+import com.deck.core.entity.misc.ChannelType
+import java.util.*
 
 public interface TeamChannel : Channel {
-    public val teamId: GenericId
+    public val teamId: GenericId?
 }
+
+public interface PartialTeamChannel : Entity {
```

Comment:
So we don't lose our objective on the core, I think we should remove most nullables and objects that we don't find necessary or even know about.